source/game_sa/Crime.cpp

```diff
@@ -1,8 +1,113 @@
 #include "StdInc.h"
 
 #include "Crime.h"
+#include "PedType.h"
+#include "eWantedLevel.h"
 
-void CCrime::ReportCrime(eCrimeType crimeType, CEntity* entity, CPed* ped2) {
-    assert(!ped2 || ped2->IsPed());
-    plugin::Call<0x532010, eCrimeType, CEntity*, CPed*>(crimeType, entity, ped2);
+void CCrime::InjectHooks() {
+    RH_ScopedClass(CCrime);
+    RH_ScopedCategoryGlobal();
+
+    RH_ScopedInstall(FindImmediateDetectionRange, 0x531FC0);
+    RH_ScopedInstall(ReportCrime, 0x532010);
+}
+
+// Android
+static const float WANTED_LEVEL_RANGES[12] = {60.0f, 14.0f, 14.0f, 14.0f, 60.0f, 60.0f, 14.0f, 14.0f, 30.0f, 0.0f, 0.0f, 0.0f};
+
+// 0x531FC0
+float CCrime::FindImmediateDetectionRange(eCrimeType CrimeType)
+{
+  if ((CrimeType - CRIME_DESTROY_HELI) > CRIME_VEHICLE_DAMAGE)
+    return 14.0;
+  else
+    return WANTED_LEVEL_RANGES[CrimeType - CRIME_DESTROY_HELI];
```

Comment:
This function doesn't look like anything it should...
```suggestion
    switch (crimeType) {
    case CRIME_TYPE_12:
    case CRIME_TYPE_16:
    case CRIME_EXPLOSION: return 60.f;
    case CRIME_TYPE_20:   return 30.f;
    default:              return 14.f;
    }
```

Reply:
Android version actually has an array like that